### ISSUE TITLE
Add support for HyperV

### ISSUE THREAD

**Issue by zamoose** [wpengine/hgv#137]
Breaking out from #10.

**Comment by ericmann** [comment]
One note as we continue to work on and test this, supporting Hyper-V will likely also require rolling in a solution for #120 at the same time.

Hyper-V doesn't support the same networking configuration that VirtualBox does under Vagrant. This means forwarded ports aren't _actually_ forwarded. It also means setting a static IP address is impossible. I've made a change in the original Hostsupdater plugin to address this, unfortunately that change breaks compatibility with VirtualBox entirely.

*In summary* The existing Hostsupdater plugin won't work in a Hyper-V situation, so while the change to support Hyper-V as a virtualization provider is quick, making sure everything works in Hyper-V is a bit more complex.

**Comment by ericmann** [comment]
Fixed by #205
